readthedocs/config/config.py

```diff
@@ -442,11 +613,6 @@ def validate(self):
         for build in self:
             build.validate()
 
-    def set_output_base(self, directory):
```

Comment:
We weren't using this, and it's the only setter. We should keep all the config object as read-only I think.